resources/ansible/roles/uploaders/tasks/main.yml

```diff
@@ -1,59 +1,68 @@
 ---
 - name: check if safe binary exists
   ansible.builtin.stat:
-    path: "{{ binary_dir}}/safe"
+    path: "{{ binary_dir }}/safe"
   register: safe_binary
 
 - name: download the safe binary
   ansible.builtin.get_url:
     url: "{{ safe_archive_url }}"
-    dest: /tmp/{{ safe_archive_filename }}
+    dest: "/tmp/{{ safe_archive_filename }}"
   when: not safe_binary.stat.exists
 
 - name: extract the safe binary to /usr/local/bin
-  become: True
   ansible.builtin.unarchive:
     src: "/tmp/{{ safe_archive_filename }}"
     dest: "{{ binary_dir }}"
-    remote_src: True
+    remote_src: true
+  become: true
   when: not safe_binary.stat.exists
 
-- name: copy upload-random-data.sh to remote
+- name: create safe users
+  ansible.builtin.user:
+    name: "safe{{ item }}"
+    shell: /bin/bash
+    state: present
+  loop: "{{ range(1, safe_uploader_instances | int + 1) | list }}"
+
+- name: copy upload-random-data.sh to remote for each safe user
   ansible.builtin.copy:
     src: upload-random-data.sh
-    dest: /home/safe/upload-random-data.sh
-    owner: safe
-    group: safe
+    dest: "/home/safe{{ item }}/upload-random-data.sh"
+    owner: "safe{{ item }}"
+    group: "safe{{ item }}"
     mode: '0744'
   become: yes
-  become_user: safe
-  when: not safe_binary.stat.exists
+  become_user: "safe{{ item }}"
+  loop: "{{ range(1, safe_uploader_instances | int + 1) | list }}"
 
-- name: obtain funds from the faucet
+# The throttle is used to avoid rate limiting from the faucet.
+# Also, when this role runs again on the upscale command, the existing uploaders will obtain
+# additional funds, but that's easier than trying to exclude them. It shouldn't do any harm.
+- name: obtain funds from the faucet for each safe user
   shell: "safe --peer {{ genesis_multiaddr }} wallet get-faucet {{ faucet_address }}"
   args:
-    chdir: /home/safe/
+    chdir: "/home/safe{{ item }}/"
     executable: /bin/bash
   become: yes
-  become_user: safe
-  # run one at a time to not be rate limited by the faucet
+  become_user: "safe{{ item }}"
   throttle: 1
-  when: not safe_binary.stat.exists
+  loop: "{{ range(1, safe_uploader_instances | int + 1) | list }}"
 
 - name: create systemd service file
   ansible.builtin.template:
     src: safe_uploader.service.j2
-    dest: /etc/systemd/system/safe_uploader.service
+    dest: "/etc/systemd/system/safe_uploader@.service"
```

Comment:
Q: does this create the service for all the` `safe* users?

Reply:
Hey, so what this does is, it creates one service definition as a template.

You then reference each service instance using this: `systemctl status safe_uploader@1`, `systemctl status safe_uploader@2` etc.

It works well when the service definition is simple.